### DISCUSSION TITLE
Cannot edit details synteny track through JEXL

### DISCUSSION THREAD

**Discussion by abjanssen** [GMOD/jbrowse-components#4540] [Q&A] [answered]
I'm trying to add some alignment tracks to our instances, and have a question about the manipulation of the Core details/Mate details/Attributes that show up if you click a feature a linear synteny track (see below).

Normally, one can manipulate the features that show up in these kinds of tracks through some JEXL code in the config file. However, I cannot find the right variables for these types of tracks.
For example, if I want to remove the mate.refName from view, I'd add "mate.refName:undefined" in the formatDetails of the track. However, that doesn't work in this case. In addition, the details that I could remove were named differently than expected. For example, "assemblyName:undefined", removed the mate.assemblyName variable, and not the 'normal ' assemblyName variable.

The crux of my question thus is, how can we manipulate the variables in the detailed description of these alignment features? Is there an overview of the variablenames that we have to use?

Thank you for your help!

![image](https://github.com/user-attachments/assets/fb7405b3-f2e6-4f37-8c43-922b3d2ce4d2)

**Comment by cmdcolin** [comment, marked as answer]
Hi there,
I found potentially a cause of this issue here https://github.com/GMOD/jbrowse-components/pull/4542

Basically the "synteny tracks" were not properly wired up to the "featureDetail" formatter callbacks

I can do a little more testing on it, but you can also try it if you want to try out that branch (e.g. `jbrowse create --branch fix_alignments_customization`)

The dot notation probably won't work but you can use 

```
"formatDetails": { 
     "feature": "jexl:{mate:{refName:'replacement stuff'}}" 
 }
```

note that this removes mate.state and mate.end and mate.assemblyName (e.g. it does not deepmerge 'replacement stuff' into the refName, leaving everything else, it replaces it so it is just mate.refName='replacement stuff' and nothing else in the details

**Comment by abjanssen** [comment]
Thank you very much, this works great!
